### PR TITLE
do not generate debug output for ndsz if dns is not enabled

### PR DESCRIPTION
**Pull request by ramaraochavali**
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/xds/debug.go [hunk 1]
```diff
@@ -781,11 +781,16 @@ func (s *DiscoveryServer) List(w http.ResponseWriter, req *http.Request) {
 // ndsz implements a status and debug interface for NDS.
 // It is mapped to /debug/ndsz on the monitor port (15014).
 func (s *DiscoveryServer) ndsz(w http.ResponseWriter, req *http.Request) {
+	proxyID, con := s.getDebugConnection(req)
+	if !con.proxy.Metadata.DNSCapture {
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/debug.go:785]: should check if con == nil

Comment (hzxuzhonghu) [reply pilot/pkg/xds/debug.go:785]: and con.proxy ==nil

**Review by hzxuzhonghu — APPROVED** [review]
lgtm, con.proxy must not be nil guaranteed by `getProxyConnection`

**Review by hzxuzhonghu — COMMENTED** [review]
/hold

@ramaraochavali please move down after existing con == nil

**Comment by ramaraochavali** [comment]
Moved

**Pull request merged**